api/src/main/java/org/openmrs/api/impl/ObsServiceImpl.java

```diff
@@ -462,7 +462,19 @@ public Integer getObservationCount(List<Person> whom, List<Encounter> encounters
 	public Obs getObsByUuid(String uuid) throws APIException {
 		return dao.getObsByUuid(uuid);
 	}
-	
+
+	/**
+	 * @see org.openmrs.api.ObsService#getRevisionObs(org.openmrs.Obs)
+	 */
+	@Transactional(readOnly = true)
+	public Obs getRevisionObs(Obs initialObs) {
+		Obs revisedObs = dao.getRevisionObs(initialObs);
+		if (revisedObs != null && revisedObs.getVoided()) {
+			revisedObs = getRevisionObs(revisedObs);
```

Comment:
The api shouldn't be the one to recursively fetch revisions, the api caller could actually be interested in the immediate revisions for some reason, so it's up to the client code to recursively do this.